04/app.js

```diff
@@ -1,87 +1,96 @@
+    this.removeUser(userName);
+  };
 
-    render() {
-        const { firstName, lastName } = this.state;
-        return (
-            <section onSubmit={ this.submitHandler }>
-                <form>
-                    <input name="firstName"
-                        value={ firstName }
-                        onChange={ this.inputChange }
-                    />
-                    <input name="lastName"
-                        value={ lastName }
-                        onChange={ this.inputChange }
-                    />
-                    <input type="submit"/>
-                </form>
-                <ul>{ this.renderUsersList() }</ul>
-            </section>
-        );
-    }
+  inputChange = (e) => {
+    const { name, value } = e.target;
+    this.setState({
+      [name]: value,
+    });
+  };
 
-    submitHandler = e => {
-        e.preventDefault();
+  render() {
+    const { firstName, lastName, searchQuery } = this.state;
+    return (
+      <section onSubmit={this.submitHandler}>
+        <form>
+          <input
+            name="firstName"
+            value={firstName}
+            onChange={this.inputChange}
+          />
+          <input name="lastName" value={lastName} onChange={this.inputChange} />
+          <input type="submit" />
+        </form>
+        <input
+          type="text"
+          name="searchQuery"
+          placeholder="Search user"
+          value={searchQuery}
+          onChange={this.inputChange}
+        />
+        <ul>{this.renderUsersList()}</ul>
+      </section>
+    );
+  }
 
-        const { firstName, lastName } = this.state;
-        if(firstName && lastName) {
-            this.addUser(`${firstName} ${lastName}`);
-            this.setState({
-                firstName: '',
-                lastName: '',
-            });
-        } else {
-            // tutaj komunikat dla użytkownika
-        }
-    }
+  submitHandler = (e) => {
+    e.preventDefault();
 
-    addUser(name) {
-        this.setState({
-            users: [...this.state.users, name],
-        });
+    const { firstName, lastName } = this.state;
+    if (firstName && lastName) {
+      this.addUser(`${firstName} ${lastName}`);
+      this.setState({
+        firstName: "",
+        lastName: "",
+      });
+    } else {
+      // tutaj komunikat dla użytkownika
     }
+  };
 
-    removeUser(name) {
-        const currUsers = this.state.users.filter(
-            user => user != name
-        );
+  addUser(name) {
+    this.setState({
+      users: [...this.state.users, name],
+    });
+  }
 
-        this.setState({
-            users: currUsers,
-        });
-    }
+  removeUser(name) {
+    const currUsers = this.state.users.filter((user) => user != name);
+
+    this.setState({
+      users: currUsers,
+    });
+  }
 }
 
-ReactDOM.render(<App/>, document.querySelector('#root'));
\ No newline at end of file
+ReactDOM.render(<App />, document.querySelector("#root"));
```

Comment:
Może dodałbyś w tym zadaniu jeszcze komunikat ilu użytkowników została ukrytych przez filtrowanie, w tym ilu nowo dodanych? Nowo dodani to tacy, którzy zostali wprowadzeni po tym jak zostały wyniki przefiltrowane ;)